### DISCUSSION TITLE
How to import existing wireguard configuration?

### DISCUSSION THREAD

**Discussion by freekode** [wg-easy/wg-easy#1460] [Q&A] [answered]
Hi,
I have running wireguard server from linuxserver.io. And would like to move to wg-easy.
I tried to manually recreate a json with data from my current server to import it in wg-easy. Json file imported successfully and I'm able to connect to the server, although there is no connection to internet. I can't open any web page.
Is there a easy way to import current configuration of wireguard? Or how to properly migrate from the 'original' wireguard server?

**Comment by kaaax0815** [comment, marked as answer]
There is currently no way and I don't think there ever will be first party support for that. (Even tho one could easily create a script)

Your best bet is to manually fill the entries in the `wg0.json`

As for the Internet problem. As I don't know your setup or your config/env I can't help without more information

**Comment by x00Pavel** [comment]
Hi team,

Looking forward to try your UI :)

I have several manually added clients already. Theoreticly would it be possible to setup your UI to create `wg1.conf` file, manually migrate clients without knowing their private keys to the UI. After this migration "migrate" the file from `wg1.conf` to `wg0.conf`. What do you think about this approach?

**Reply by kaaax0815** [reply]
wg-easy won't read the wg0.conf file. everything has to be stored in the database

**Reply by dablenparty** [reply]
But... _why_? This is what's keeping me from using `wg-easy`: Wireguard already has its own configuration, which `wg-easy` has to use to get up and running anyway, so masking that with your own config implementation is not only more work for you, but also makes the config less accessible to users and migrating both to and from other Wireguard services becomes far more difficult. Speaking of migrating, you claim you don't plan on supporting migration from the standard Wireguard configuration to your proprietary config yourself, but then turn around to claim it could be "easily" done. I guess I just don't understand the logic here.

If you can tell me how I can take my existing Wireguard config and use that with `wg-easy`, or better yet, tell me how I can use the `wg-easy` UI with a Wireguard instance running in a _separate_ container (e.g. [LinuxServer.io's image](https://docs.linuxserver.io/images/docker-wireguard/) is a rootless container), I'll happily switch right now. But I'm not going to spend _my_ time reading through _your_ source code so that I can convert _my_ config to _your_ **proprietary** database schema, _especially_ when this migration isn't even documented.

**Reply by kaaax0815** [reply]
> But... _why_? 

Because nobody implemented it

> Speaking of migrating, you claim you don't plan on supporting migration from the standard Wireguard configuration to your proprietary config yourself, but then turn around to claim it could be "easily" done. I guess I just don't understand the logic here.

One could implement the logic if needed. And it's fairly easy to manually migrate it.

And I wouldn't call the config proprietary as it's literally open source

> But I'm not going to spend _my_ time reading through _your_ source code so that I can convert _my_ config to _your_ **proprietary** database schema, _especially_ when this migration isn't even documented.

Nobody asked you that. And nobody is forcing you to use wg-easy. But deploying it with the other container you mentioned is just not the goal of this project.

**Reply by dablenparty** [reply]
> Because nobody implemented it.

Nobody implemented  _what_? What does this mean?

> One could implement the logic if needed. And it's fairly easy to manually migrate it.

Again, you’re not really answering me. You’re just repeating what you said before. I want to know why, if the logic is so easy, have you not implemented it yourself? And why is this even necessary?

> And I wouldn't call the config proprietary as it's literally open source

Something can be proprietary and open source, they’re not mutually exclusive, but I’ll admit it probably wasn’t the _best_ choice of word. What I’m trying to say is you’ve made a config format that works with your program _only_. I can’t copy a wg-easy config to another Wireguard instance or vice-versa, which is different from every other Wireguard wrapper I’ve seen so far. This kind of lock-in is something I usually only see commercially, hence why I called it proprietary. What I don’t understand is why you made your own config when Wireguard has a perfectly understandable and usable config already?

> Nobody asked you that. And nobody is forcing you to use wg-easy.

No, nobody asked me that, but _you_ were asked how to import an existing Wireguard config and responded with:

> There is currently no way and I don't think there ever will be first party support for that. (Even tho one could easily create a script)
>
> Your best bet is to manually fill the entries in the wg0.json

where you straight up say to do it manually. If you’re talking specifically about writing the script, instead of saying “write it yourself,” you're saying “I won’t write it myself” which is the same thing, just inverted. 

> But deploying it with the other container you mentioned is just not the goal of this project.

Thank you for answering that part.

**Reply by kaaax0815** [reply]
> Nobody implemented  _what_? What does this mean?

The ability to migrate a wireguard config to wg-easy

> Again, you’re not really answering me. You’re just repeating what you said before. I want to know why, if the logic is so easy, have you not implemented it yourself? And why is this even necessary?

No I have not. Because a wireguard config is not enough for wg-easy

> What I don’t understand is why you made your own config when Wireguard has a perfectly understandable and usable config already?

Because there is other stuff to configure, the host the clients should connect to, usernames, ...

> where you straight up say to do it manually. If you’re talking specifically about writing the script, instead of saying “write it yourself,” you're saying “I won’t write it myself” which is the same thing, just inverted. 

Yes. Because just because someone requests a feature doesn't mean it will be implemented. People have limited time. 

If you want it, you either ask for it, and if it won't be implemented you have to do it yourself. That's how non commercial projects usually work
